src/micro_http/src/connection.rs

```diff
@@ -297,7 +311,9 @@ impl<T: Read + Write> HttpConnection<T> {
 
     /// Tries to write the first available response to the provided stream.
     /// Meant to be used only with non-blocking streams and an `EPOLL` structure.
-    /// Should be called whenever an `EPOLLOUT` event is signaled.
+    /// Should be called whenever an `EPOLLOUT` event is signaled. If connection closed
```

Comment:
Actually, responses are discarded when **no bytes were written to the stream or in case an error appeared while doing so**. `connection_closed` is just a flag for marking that the response queue should be emptied.